async.go

```diff
@@ -17,23 +20,49 @@ import (
 // before being pushed to Core Data.
 func processAsyncResults() {
 	for !svc.stopped {
-		readings := make([]models.Reading, 0, svc.c.Device.MaxCmdOps)
-		cr := <-svc.asyncCh
+		acv := <-svc.asyncCh
+		readings := make([]models.Reading, 0, len(acv.CommandValues))
 
-		// get the device resource associated with the rsp.RO
-		do := pc.getDeviceObjectByName(cr.DeviceName, cr.RO)
+		deviceName := acv.DeviceName
+		device, ok := cache.Devices().ForName(deviceName)
+		if !ok {
+			common.LogCli.Error(fmt.Sprintf("processAsyncResults - recieved Device %s not found in cache", deviceName))
+			continue
+		}
 
-		_ = cr.TransformResult(do.Properties.Value)
+		for _, cv := range acv.CommandValues {
+			// get the device resource associated with the rsp.RO
+			do, ok := cache.Profiles().DeviceObject(device.Profile.Name, cv.RO.Object)
+			if !ok {
+				common.LogCli.Error(fmt.Sprintf("processAsyncResults - Device Resource %s not found in Device %s", cv.RO.Object, deviceName))
+				continue
+			}
 
-		reading := cr.Reading(cr.DeviceName, do.Name)
-		readings = append(readings, *reading)
+			if common.CurrentConfig.Device.DataTransform {
+				err := transformer.TransformReadResult(cv, do.Properties.Value)
+				if err != nil {
+					common.LogCli.Error(fmt.Sprintf("CommandValue (%s) transformed failed: %v", cv.String(), err))
+				}
+			}
 
-		// push to Core Data
-		event := &models.Event{Device: cr.DeviceName, Readings: readings}
-		_, err := svc.ec.Add(event)
-		if err != nil {
-			msg := fmt.Sprintf("internal error; failed to push event for dev: %s to CoreData: %s", cr.DeviceName, err)
-			svc.lc.Error(msg)
+			err := transformer.CheckAssertion(cv, do.Properties.Value.Assertion, &device)
+			if err != nil {
+				common.LogCli.Error(fmt.Sprintf("Assertion failed for Device Resource: %s, with value: %s", cv.String(), err))
```

Comment:
the first two log messages use the function name as a prefix, but this one doesn't. also, if the assertion fails, why push?

Reply:
you are right, modified as the previous conversation